### PR TITLE
Create sessions for each user, save their responses and resume sessions

### LINKED ISSUE #31: Store the response on answer submission and load past answers upon reloading page

### PR DESCRIPTION
**Pull request by dalmia**
Fixes #31 

Linked to: https://github.com/avantifellows/quiz-backend/pull/24

## Summary

- adds a check for `userId` to have been passed as a query param in the URL - if not, the user is taken to a 403 access denied page. So, before this PR, you were able to access the quiz by visiting: `http://localhost:8080/62540adb0f748c8e206c1612`. Now, you would have to pass in the `userId` too: `http://localhost:8080/62540adb0f748c8e206c1612?userId=10`.

- adds API calls to create session and update a session's answer
- creates session when the Player page is opened - changes button text based on whether it is their first session or are they resuming and restores the answers that they have previously given to the questions.
- adds a few new types

## Test Plan

- [x] Test Responsiveness
  - [ ] Laptop (1200px)
  - [ ] Tablet (760px)
  - [ ] Phone (320px)
- [x] Cross-Browser Testing
  - [ ] Chrome
  - [ ] Firefox
- [ ] ~Local Language Support~
- [x] Hygiene and Housekeeping
  - [x] Self-review
  - [x] Comments have been added appropriately
  - [ ] ~Check for bundle size [here](https://bundlephobia.com/) if adding a package~
  - [x] Added relevant details like Labels/Projects/Milestones etc.
  - [ ] ~If adding or removing any environment variable, update `docs/ENV.md`, `.env.example` and the Github workflows.~
- [ ] Testing
  - [x] Wrote tests
  - [x] Tested locally
  - [x] Tested on staging
  - [ ] Tested on an actual physical phone
  - [ ] Tested on production
- [x] Lighthouse Checks
  - [ ] ~Images have `alt` attributes~
  - [ ] ~Any `` tags have `width` and `height` specified~
  - [ ] ~Any `target="_blank"` links have `rel="noopener"`~
  - [x] Only SVGs are used as images. If PNGs are used, their size has been optimised.
  - [ ] ~Any SVG buttons without text have their `aria-label` attributes set~

### CODE REVIEW

File: src/services/API/Session.ts [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+} from "../../types";
+
+export default {
+  /**
+   * returns the details for a quiz
+   * @param {string} quizId - id of the quiz for which the session is to be created
+   * @param {string} userId - id of the user for which the session is to be created
+   * @returns {Promise<SessionAPIResponse>} data corresponding to the session
+   */
+  async createSession(
```
Comment (pritamps) [inline src/services/API/Session.ts:16]: To maintain consistency across our "products", call this component `UserQuizSession`?

Comment (dalmia) [reply src/services/API/Session.ts:16]: Even within the quiz repo? It always includes the Quiz part implicitly no?

**Comment by deepansh96** [comment]
Even if I have no answered any questions, but I just came to the quiz link, it'll create a session for me. And because of that, the next time I open, it'll show me "Resume" instead of "let's start"
Is that fine?
Shouldn't it show "Resume" only if I have answered some question?

**Review by deepansh96 — CHANGES_REQUESTED** [review]

**Comment by sudheshna-donthineni** [comment]
> Even if I have no answered any questions, but I just came to the quiz link, it'll create a session for me. And because of that, the next time I open, it'll show me "Resume" instead of "let's start" Is that fine? Shouldn't it show "Resume" only if I have answered some question?

Not necessarily, no? I have seen both cases happen in products.

**Review by deepansh96 — APPROVED** [review]

File: src/views/Player.vue [hunk 2]
```diff
@@ -16,25 +17,48 @@
         :questions="questions"
         v-model:currentQuestionIndex="currentQuestionIndex"
         v-model:responses="responses"
+        @submit-question="submitQuestion"
```
Comment (sudheshna-donthineni) [inline src/views/Player.vue:20]: is this some TS formatting? The kebab case for method handlers? I always wrote them like `@submitQuestion`.

**Pull request merged**